src/power_process.c

```diff
@@ -667,4 +668,78 @@ void remove_explored_flags_for_power_sight(struct PlayerInfo *player)
         }
     }
 }
+
+void process_timebomb(struct Thing *creatng)
+{
+    SYNCDBG(18,"Starting");
+    if (!creature_affected_by_spell(creatng, SplK_TimeBomb)) {
+        return;
+    }
+    update_creature_speed(creatng);
+    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
+    struct Thing* timetng = thing_get(cctrl->timebomb_countdown_id);
+    if (thing_is_invalid(timetng))
+    {
+        if ((cctrl->timebomb_countdown % game.num_fps) == 0)
+        {
+            long time = (cctrl->timebomb_countdown / game.num_fps);
+            timetng = create_price_effect(&creatng->mappos, creatng->owner, time);
+            cctrl->timebomb_countdown_id = timetng->index;
+            thing_play_sample(creatng, 853, NORMAL_PITCH, 0, 3, 0, 2, FULL_LOUDNESS);
+        }
+    }
+    else
+    {
+        if (timetng->health <= 0)
+        {
+            delete_thing_structure(timetng, 0);
+            cctrl->timebomb_countdown_id = 0;
+        }
+        else
+        {
+            struct Coord3d pos;
+            pos.x.val = creatng->mappos.x.val;
+            pos.y.val = creatng->mappos.y.val;
+            pos.z.val = timetng->mappos.z.val;
+            move_thing_in_map(timetng, &pos);
+        }
+    }
+    struct Thing* trgtng = thing_get(cctrl->timebomb_target_id);
+    if (!thing_is_invalid(trgtng))
+    {
+        if ( (creatng->mappos.x.stl.num == trgtng->mappos.x.stl.num) && (creatng->mappos.y.stl.num == trgtng->mappos.y.stl.num) && (creatng->mappos.z.stl.num == trgtng->mappos.z.stl.num) )
```

Comment:
I noticed an issue with timebomb not exploding on flying creatures sometimes, could be because of this.

Reply:
Is it better now?